blockstack/stacks-blockchain-api
# Support sponsored transactions

Depends on https://github.com/blockstack/stacks-transactions-js/pull/99

This PR updates the sidecar to support processing and returning sponsored tx data. 

### Testing

The `/sidecar/v1/debug/broadcast` endpoints now have a checkbox to create and broadcast a sponsored transaction. 

For example: 
* Visit `http://localhost:3999/sidecar/v1/debug/broadcast/contract-deploy`
* Check `[x] Create sponsored transaction` and click Submit
* The tx response will then contain sponsored tx info, for example: 
```js
// localhost:3999/sidecar/v1/tx/0x2537c7090545c01db3f86300efb75b72e07695c6d4bf5358d218242d2da859a1
{
  ...
  "sponsored": true,
  "sponsor_address": "STRYYQQ9M8KAF4NS7WNZQYY59X93XEKR31JP64CP",
  ...
}
```
